platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateRoot.java

```diff
@@ -1005,8 +1005,20 @@ public SwirldState getSwirldState() {
      */
     @NonNull
     @Override
-    public PlatformStateAccessor getPlatformState() {
-        return !isImmutable() ? writablePlatformStateStore() : readablePlatformStateStore();
+    public PlatformStateAccessor getReadablePlatformState() {
```

Comment:
How much work would it be to make this method return `ReadablePlatformStateStore` and `getWritable()` below return `WritablePlatformStateStore`?

Reply:
`PlatformStateAccessor` is an abstraction layer that hides the implementation details of the platform state. This interface is implemented by two classes - `com.swirlds.platform.state.State` and `com.swirlds.platform.state.MerkleStateRoot`. In case of the latter, the platform state is backed by `ReadablePlatformStateStore` and `WritablePlatformStateStore`, in case of the former it is not. 

That is, the question is not in the amount of work. The question is in the possibility of this option. 

I also considered a different approach (as we discussed) - accessing the platform state via `com.swirlds.state.State` -> `com.swirlds.state.spi.ReadableStates#getSingleton`. But the solution turned out to be rather ugly. In each case I have to explicitly case and instance of `MerkleStateRoot` to `com.swirlds.state.State` because throughout the code we work with via `com.swirlds.platform.state.MerkleRoot` interface. 

I thought that introducing two methods `getReadablePlatformState` and `getWritablePlatformState` to `PlatformStateAccessor` is a lesser evil. However, I'm open to be persuaded otherwise.

Reply:
Ah good point. 💯